### DISCUSSION TITLE
Warn if "Photon cross sections=" is incorrectly specified in the media definition block?

### DISCUSSION THREAD

**Discussion by dworogers** [nrc-cnrc/EGSnrc#651] [Ideas]
I thought it reasonable that `Photon cross sections=` could be specified in the pegsless media definition block and EGSnrc flags no error if included there. But it has no effect. `Photon cross sections=` can only be specified in the MC transport parameter block. Not sure the best approach but as a minimum, if not changed to allow it in the media definition block, there should be a warning and the code stop if mistakenly entered there.

**Comment by ftessier** [comment]
I disagree. The media definition input block is in my view strictly reserved for pegs related inputs, and I don't want to blur that line by including transport parameters there. On the subject of input validation, it would be great to have a formal grammar to validate `.egsinp` files (easier if we moved to a recognized standard markup language such as [TOML](https://github.com/toml-lang/toml) or JSON). I don't see why we would single out `Photon cross sections=` to check if it is in the right block. We _could_ check every key to see if it is potentially in the wrong block, but I think this approach rapidly spirals out of control (as the code evolves) compared to a clear reference on the proper input format (not saying that this clear reference exists at the moment 😄 ).

**Reply by dworogers** [reply]
This is a pretty fine point. For example, pirs701 includes as an example selecting the NRC brems option which is not available via the egs_gui inputs to pegs4.  Is that a media input or a transport input?  So distinguishing between photon and electron specification in the media definition block seems misleading at best. But the solution may be to have the manual make clear that this section is only about electron cross sections (if that is the case) and photon cross sections are selected in the transport block.

I had farmed my input file out to 5 power users of EGS when I was getting some inconsistent results and not one of them identified that this entry in the media block was wrong. It may be worthwhile just having a check on this class of fairly natural mistakes or at least make it very clear in the manual that things like selecting the photon cross section belongs on the transport parameters block. Perhaps removing the brems option as an example in the manual would help too since it doesn't really define the medium.

**Reply by ftessier** [reply]
Just to clarify the discussion: the `bremsstrahlung correction = NRC` pegs4 input (`IAPRIM=1`) in the media input block is distinct from the `Brems cross sections= [ BH | NIST | NRC ]` (the `ibr_nist` parameter) in the Monte Carlo transport parameter block. If I understand correctly, the first one scales the integrated brems cross section for a material to match the NIST stopping powers, while the second specifies which differential brems cross section model to use for transport. In fact, unless one is trying to recover older EGS4 data, there is no obvious reason _not_ to select `bremsstrahlung correction = NRC`. This is the default, and this key should not normally appear in the media input. In fact I suggest we _remove_ it in the examples provided in pirs701, to avoid confusion.

**Comment by rtownson** [comment]
In the new egsinp editor I'm working on integrating into egs_view, invalid or unknown inputs will have a red squiggle under them (like a spelling error), so that will help resolve the issue!
